tornado_rest_jsonapi/tests/resource_handlers.py

```diff
@@ -16,36 +16,37 @@ class WorkingDataLayer(BaseDataLayer):
     id = 0
 
     @gen.coroutine
-    def create_object(self, data, **kwargs):
+    def create_object(self, data, view_kwargs):
         id = str(type(self).id)
         data["id"] = id
         self.collection[id] = data
         type(self).id += 1
-        return id
+        return data
 
     @gen.coroutine
-    def retrieve_object(self, identifier, **kwargs):
+    def get_object(self, kwargs):
+        identifier = kwargs.get("id")
         if identifier not in self.collection:
             raise exceptions.ObjectNotFound()
 
         return self.collection[identifier]
 
     @gen.coroutine
-    def update_object(self, identifier, data, **kwargs):
-        if identifier not in self.collection:
-            raise exceptions.ObjectNotFound()
-
-        self.collection[identifier].update(data)
+    def update_object(self, obj, data, view_kwargs):
+        obj.update(data)
+        return True
 
     @gen.coroutine
-    def delete_object(self, identifier, **kwargs):
+    def delete_object(self, obj, view_kwargs):
+        identifier = view_kwargs.get("id")
```

Comment:
There's technically no guarantee there's an id in the resulting object. That object is obtained from a get operation, and it's the model object. you could, technically, create it on the spot. I agree it would be a very dumb usage, but that's the API we have to work with. I don't agree with most of the choices they made, but honestly I could not design anything better and equally flexible.